packages/ai-chat/src/common/chat-session-naming-service.ts

```diff
@@ -0,0 +1,122 @@
+        'IMPORTANT: Your answer MUST ONLY CONTAIN THE PROPOSED NAME and must not be preceded or succeeded with any other text.' +
+        '\n\nOther session names:\n{{listOfSessionNames}}' +
+        '\n\nConversation:\n{{conversation}}',
+};
+
+@injectable()
+export class ChatSessionNamingService implements Agent {
+    id = 'chat-session-naming-service';
+    name = 'Chat Session Naming Service';
+    description = 'Service to generate chat session names';
+    variables = [];
+    promptTemplates: PromptTemplate[] = [CHAT_SESSION_NAMING_PROMPT];
+    languageModelRequirements: LanguageModelRequirement[] = [{
+        purpose: 'chat-session-naming',
+        identifier: 'openai/gpt-4o',
+    }];
+    agentSpecificVariables = [
+        { name: 'conversation', usedInPrompt: true, description: 'The content of the chat conversation.' },
+        { name: 'listOfSessionNames', usedInPrompt: true, description: 'The list of existing session names.' }
+    ];
+    functions = [];
+
+    @inject(LanguageModelRegistry)
+    protected readonly lmRegistry: LanguageModelRegistry;
+
+    @inject(CommunicationRecordingService)
+    protected recordingService: CommunicationRecordingService;
+
+    @inject(PromptService)
+    protected promptService: PromptService;
+
+    async generateChatSessionName(chatSession: ChatSession, otherNames: string[]): Promise<string> {
+        const lm = await this.lmRegistry.selectLanguageModel({ agent: this.id, ...this.languageModelRequirements[0] });
+        if (!lm) {
+            throw new Error('No language model found for chat session naming');
+        }
+        if (chatSession.model.getRequests().length < 1) {
+            throw new Error('No chat request available to generate chat session name');
+        }
+
+        const conversation = chatSession.model.getRequests()
+            .map(req => `<user>${req.request.text}</user>` +
+                (req.response.response ? `<assistant>${req.response.response.asString()}</assistant>` : ''))
+            .join('\n\n');
+        const listOfSessionNames = otherNames.map(name => name).join(', ');
+
+        const prompt = await this.promptService.getPrompt(CHAT_SESSION_NAMING_PROMPT.id, { conversation, listOfSessionNames });
+        const message = prompt?.text;
+        if (!message) {
+            throw new Error('Unable to create prompt message for generating chat session name');
+        }
+
+        const request: LanguageModelRequest = {
+            messages: [{
+                actor: 'user',
+                query: message,
+                type: 'text'
+            }]
+        };
+
+        const sessionId = generateUuid();
+        const requestId = generateUuid();
+        this.recordingService.recordRequest({
+            agentId: this.id,
+            sessionId,
+            requestId,
+            ...request
+        });
+
+        const result = await lm.request(request);
+        const response = await getTextOfResponse(result);
```

Comment:
Should we do some sanity checks, i.e. check for markers, length etc?

Reply:
Good question, I didn't encounter issues, but I guess, you're right, we should remove at least line breaks and limit to a certain number of characters.